pkg/k8sutils/k8sutils.go

```diff
@@ -29,55 +31,34 @@ func GetK8sClient(config *rest.Config) (kubernetes.Interface, error) {
 
 }
 
-func GetDeploymentStatus(k8sclient kubernetes.Interface, namespace string, name string) bool {
+func GetDeploymentStatus(operatorClient client.Client, namespace string, name string) bool {
 
-	if k8sclient == nil {
-		logging.Log.Info("Failed to get Deployment: No k8sclient initiated")
-		return false
-	}
+	var timeout time.Duration = 420
+	var retryInterval time.Duration = 10
 
-	api := k8sclient.AppsV1()
-	var timeout int64 = 420
-	listOptions := metav1.ListOptions{
-		FieldSelector:  fields.OneTermEqualSelector("metadata.name", name).String(),
-		TimeoutSeconds: &timeout,
-	}
-	watcher, err := api.Deployments(namespace).Watch(listOptions)
-	if err != nil {
-		logging.Log.Error(err, "An error occurred")
-	}
-	ch := watcher.ResultChan()
-	logging.Log.Info("Waiting for deployment", "Namespace", namespace, "Name", name, "Timeout", timeout)
-	for event := range ch {
-		dc, ok := event.Object.(*appsv1.Deployment)
-		if !ok {
-			logging.Log.Error(err, "Unexpected type")
-		}
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
```

Comment:
Ripped out the need for the K8Client and instead am using the operator sdk client

Reply:
This is good, I will go and incorporate the same into my [native k8s PR](https://github.com/redhat-cop/quay-operator/pull/125/files)

Reply:
This does not work as the durations are in ms and fail immediately. Plus update accordingly

Reply:
updated

Reply:
timeout duration is not correct. please change 42->420

Reply:
Done, along with fixed a false positive with the polling loop.